demo/admin/src/products/categories/ProductCategoriesPage.tsx

```diff
@@ -45,7 +52,40 @@ const ProductCategoriesPage: React.FC = () => {
                     {(selectedId) => (
                         <SaveBoundary>
                             <FormToolbar />
-                            <ProductCategoryForm id={selectedId} />
+                            <MainContent fullHeight>
+                                <RouterTabs>
+                                    <RouterTab
+                                        forceRender={true}
+                                        path=""
+                                        label={intl.formatMessage({ id: "products.editProductCategory.formTab", defaultMessage: "Product category" })}
+                                    >
+                                        <ProductCategoryForm id={selectedId} />
+                                    </RouterTab>
+                                    <RouterTab
+                                        forceRender={true}
+                                        path="/assigned-products"
+                                        label={intl.formatMessage({
+                                            id: "products.editProductCategory.assignedProducts",
+                                            defaultMessage: "Assigned Products",
+                                        })}
+                                    >
+                                        <Box sx={{ height: "100vh" }}>
+                                            <ProductsGrid filter={{ category: { equal: selectedId } }} />
+                                            {/* TODO change button behaviour and open edit-dialog, open discussion https://github.com/vivid-planet/comet/pull/2171 */}
+                                            <EditDialog
```

Comment:
Shouldn't the EditDialog be defined inside AssignProductsForm?

Reply:
In my opinion no, why should a form include it's dialog? a form should be usable everywhere. a form should not know or define where it's rendered. Maybe I could add an additional component "AssignProductsDialogForm" combining EditDialog and AssignProductsForm. But I don't think moving this 5 lines of code would bring any benefit.